src/org/opensolaris/opengrok/index/IndexDatabase.java

```diff
@@ -447,14 +456,25 @@ public void update() throws IOException, HistoryException {
                     reader.close();
                 }
             }
+
+            try {
+                writer.commit();
+            } catch (IOException e) {
+                writerException = e;
+                LOGGER.log(Level.WARNING,
+                    "An error occured while committing writer", e);
+            }
         } finally {
-            if (writer != null) {
-                try {
-                    writer.prepareCommit();
-                    writer.commit();
-                    writer.close();
-                } catch (IOException e) {
-                    LOGGER.log(Level.WARNING, "An error occured while closing writer", e);
+            try {
+                if (writer != null) writer.close();
+            } catch (IOException e) {
+                if (writerException != null) writerException = e;
```

Comment:
Oops this should be ==. (I had to recreate this patch after accidentally git-resetting my workspace 😀.)